Dev Environments Option

davegw opened the pull request:
- Adds configuration support for notifying based on development environment (similar to the HB ruby gem)

davegw commented:
Happy to update README as well if this feature gets adopted.

joshuap commented:
@davegw nice! Sure, go ahead and update the README. Actually, could you also remove mention of sending "otherMetadata" in the server object? I'd like to stop recommending that because our API spec doesn't officially support random keys in that object. The `context` option should be used for arbitrary metadata. In a future version I'll probably make it impossible to set arbitrary keys on the server object. [These are the keys we actually support](https://gist.github.com/joshuap/94901ba378fd09a783be#file-notices-json-L237).

davegw commented:
@joshuap awesome! Updated the README to include reference to the `development_environments` option and removed mention of the 'otherMetadata' option from the server object. Thanks!
